x-pack/plugins/apm/server/lib/observability_overview/has_data.ts

```diff
@@ -28,9 +28,15 @@ export function getHasData({ setup }: { setup: Setup }) {
       };
 
       const response = await apmEventClient.search(params);
-      return response.hits.total.value > 0;
+      return {
+        hasData: response.hits.total.value > 0,
+        indices: setup.indices['apm_oss.transactionIndices']!,
```

Comment:
This seems a little odd to return, since it's a `hasData` endpoint. If we really want to do it, shouldn't this include the indices that are being queried to determine if there's data?

Eg:
```suggestion
        indices: [
          setup.indices[ProcessorEvent.transaction], 
          setup.indices[ProcessorEvent.error], 
          setup.indices[ProcessorEvent.metric]
        ],       
```

Reply:
ideally i think we should return info like this from all endpoints. I will modify it to return whole APM indices settings, instead of just transaction.

Reply:
Why should every API return this info? We already have an API (`GET /api/apm/settings/apm-index-settings`) where you can get this.

Reply:
i meant in a sense of visiblity perspective, the way we are doing that inspect stuff for debugging purpose, i think it's no harm if we return indices information as an API response, it instanlty give you a view, against which indices that API got executed.